File: Data Structures & Algorithms/top-k-elements-in-list/submission-8.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = {}
        for _, n in enumerate(nums):
            count[n] = 1 + count.get(n, 0)
        
        bucket = [[] for _ in range(len(nums) + 1)]
        for n, freq in count.items():
            bucket[freq].append(n)
        
        res = []
        for i in range(len(bucket) - 1, 0, -1):
            for j in bucket[i]:
                res.append(j)
                if len(res) == k:
                    return res

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-8.rs
impl Solution {
    pub fn find_min(nums: Vec<i32>) -> i32 {
        let (mut l, mut r) = (0, nums.len() - 1);
        while l < r {
            let mut m = (l + r) / 2;
            if nums[m] > nums[r] {
                l = m + 1
            } else {
                r = m
            }
        }
        nums[l]
    }
}
